imperial_coldfront_plugin/views.py

```diff
@@ -161,3 +162,30 @@ def accept_group_invite(request: HttpRequest, token: str) -> HttpResponse:
         context={"inviter": group.owner, "group": group.name},
         template_name="imperial_coldfront_plugin/accept_group_invite.html",
     )
+
+
+@login_required
+def get_active_users(request: HttpRequest) -> HttpResponse:
+    """Get the active users in unix passwd format.
+
+    TODO: the UnixUID must exist for each user in a group, this needs to be ensured.
+
+    Args:
+        request: The HTTP request object containing metadata about the request.
+    """
+    passwd = ""
+    format_str = (
+        "{user.username}:x:{uid.identifier}:{group.gid}:{user.first_name} "
+        "{user.last_name}:/rds/general/user/{user.username}/home:/bin/bash\n"
+    )
+    qs = User.objects.filter(groupmembership__member__isnull=False).distinct()
+    for user in qs:
+        uid = UnixUID.objects.get(user=user)
+        group = GroupMembership.objects.get(member=user).group
+        passwd += format_str.format(user=user, uid=uid, group=group)
+    for user in User.objects.filter(userprofile__is_pi=True).distinct().difference(qs):
+        uid = UnixUID.objects.get(user=user)
+        group = ResearchGroup.objects.get(owner=user)
+        passwd += format_str.format(user=user, uid=uid, group=group)
```

Comment:
This is effectively generating and returning a new `passwd` file for *everyone* registered at Imperial and associated with a group, right? Besides it being converted to JSON right now.